draft-dt-tls-external-psk-guidance.md

```diff
@@ -274,6 +256,22 @@ pair-wise shared keys to achieve this. As another example, some systems require
+2. Each PSK MUST NOT be shared between with more than two logical nodes. As a result, an agent that acts as both a client and a server MUST use distinct PSKs when acting as the client from when it is acting as the server. This prevents redirection attacks.
+3. Nodes SHOULD use external PSK importers {{!I-D.ietf-tls-external-psk-importer}} when configuring PSKs for a pair of TLS client and server. {{!RFC8446}} mandates that each external PSK can only be used with a single hash function. {{!I-D.ietf-tls-external-psk-importer}} overcomes this limitation by producing a set of candidate PSKs, each of which are 
+bound to a specific KDF. 
+4. {{!I-D.ietf-tls-external-psk-importer}} also helps in mitigating selfie style reflection attacks if a pair of client and server can switch roles. This is achieved by correctly using the node identifiers in the ImportedIdentity.context construct as explained in Appendix B of {{!I-D.ietf-tls-external-psk-importer}}. When a PSK is shared between a client and server that can switch roles, it is not necessary to know the identifier(s) of the other endpoint. Instead, it is sufficient to check that the identifier of the other endpoint in context is not equal to any of one’s own identifiers. To simplify implementation of this check, it is RECOMMENDED that each endpoint selects one globally unique identifier and uses it in all PSK handshakes. The unique identifier can, for example, be one of its MAC addresses or a 32-byte random number. Including node identifiers in the ImportedIdentity.context construct also ensures that condition (2) is met.
+5. It is NOT RECOMMENDED to share the same PSK between more than one client and server. However, as discussed in {{use-cases}}, there are application scenarios that may rely on sharing the same PSK among multiple nodes. {{!I-D.ietf-tls-external-psk-importer}} MUST be used in these application scenarios to derive unique keys for each logical node pair. Using node identifiers in the ImportedIdentity.context construct can again help nodes to derive unique keys from the global PSK shared among all nodes. Unlike the case of a server and client pair that can switch roles, when the same PSK is shared among many nodes, the comparison with one’s own identifiers will only prevent selfie attacks but not malicious rerouting of the connection to another group member. To prevent malicious rerouting in groups, each endpoint needs to know the identifier of the other endpoint with which they want to connect and compare it with the other endpoint’s identifier used in ImportedIdentity.context. Of course, this only prevents attacks by non-members; the endpoints that share the group key can always impersonate each other. 
```

Comment:
It's really hard for me to pull out the requirement from this paragraph. Can we please simplify it, or remove it altogether?

Reply:
It is quite simple. We don't recommend sharing PSKs among many nodes. If someone's use case still wants that, then they should use external PSK importers. However, that by itself is not enough. Nodes should know the other end points identifier to prevent rerouting of attacks. And of course, endpoints that know the PSK can always impersonate.

Reply:
This is a lot of text to simply say, "If you share PSKs, make sure you also include some other form of unique identifier." Can we please simplify it?